File: services/frontend/src/types/schedule.ts
export interface Session {
  session_name: string;
  start_time: string;
  end_time: string;
  capacity?: number;
  lanes?: number;
}

export interface ScheduleDetail {
  day_type: string; // "평일", "토요일", "일요일"
  season: string; // "하절기", "동절기", ""
  season_months: string; // "3~10월", "11~2월", ""
  sessions: Session[];
}

export interface Fee {
  category: string;
  price: number;
  note: string;
}

export interface ClosureInfo {
  is_closed: boolean;
  closure_type: 'monthly' | 'partial';
  reason?: string;
  specific_dates?: number;
  regular_closures?: {
    day_of_week: string;
    week_pattern: string | null;
    reason: string | null;
  }[];
}

export interface Schedule {
  facility_id: number;
  facility_name: string;
  valid_month: string;
  schedules: ScheduleDetail[];
  fees?: Fee[];
  notes?: string[];
  source_url?: string;
  created_at?: string;
  closure_info?: ClosureInfo;
}

export interface Facility {
  facility_name: string;
  latest_month: string;
  schedule_count: number;
}

export interface DailySchedule {
  facility_id: number;
  facility_name: string;
  address?: string | null;
  website_url?: string | null;
  date: string;
  day_type: string;
  season: string;
  valid_month: string;
  sessions: Session[];
  source_url?: string;
  notes?: string;
  is_closed: boolean;
  closure_reason?: string | null;
}

export interface CalendarData {
  year: number;
  month: number;
  schedules: Schedule[];
}

File: services/api/app/schemas/schedule.py
"""
Schedule related schemas
"""
from typing import List, Optional
from pydantic import BaseModel


class SessionSchema(BaseModel):
    """수영 세션 스키마"""
    session_name: str
    start_time: str
    end_time: str
    capacity: Optional[int] = None
    lanes: Optional[int] = None
    applicable_days: Optional[str] = None  # 적용 요일 (NULL=전체, "수"=수요일만)


class ScheduleDetailSchema(BaseModel):
    """요일별 스케줄 스키마"""
    day_type: str  # "평일", "토요일", "일요일"
    season: str  # "하절기", "동절기", ""
    season_months: str  # "3~10월", "11~2월", ""
    sessions: List[SessionSchema]


class FeeSchema(BaseModel):
    """이용료 스키마"""
    category: str
    price: int
    note: str


class ScheduleResponse(BaseModel):
    """스케줄 응답 스키마"""
    id: int
    facility_name: str
    valid_month: str
    schedules: List[ScheduleDetailSchema]
    fees: List[FeeSchema]
    notes: List[str]
    source_url: Optional[str] = None
    created_at: str

    class Config:
        from_attributes = True
